### PR TITLE
Retry handling

### PR DESCRIPTION
**Pull request by dbart01**
### What this does

Renames and scopes client classes under `Graph` namespace. Apparently, `git` can't handle the rename well so the diff appears as a delete / add rather than a modification.

Aside from that, the point of interest in this PR is the `Graph.RetryHandler` that allows clients to provide a `condition` upon which the request will continue retrying until the `condition` returns `false`. The primary purpose of this functionality is to facilitate an easier adoption of polling APIs for fetching shipping rates and monitoring checkout status.

**NOTE: Tests are failing because they aren't running the latest version of Swift that allows nesting in generic types and other features that are required by this PR. Please disregard the failures**

### CODE REVIEW

File: Buy/Graph.RetryHandler.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public extension Graph {
+    
+    public struct RetryHandler<R: GraphQL.AbstractResponse> {
```
Comment (davidmuzi) [inline Buy/Graph.RetryHandler.swift]: 🆒

File: Buy/Graph.Client.swift [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        
+        // ----------------------------------
+        //  MARK: - Init -
+        //
+        public init(shopDomain: String, apiKey: String, session: URLSession = URLSession(configuration: URLSessionConfiguration.default)) {
+            
+            let shopURL  = Client.urlFor(shopDomain)
+            self.apiURL  = Client.urlFor(shopDomain, path: "/api/graphql")
+            self.session = session
+            self.headers = [
```
Comment (davidmuzi) [inline Buy/Graph.Client.swift]: To be consistent with other Buy SDKs, should send:
`X-SDK-Version` which is just version, then `X-SDK-Variant` which is your free form `ios` or `android`

Comment (davidmuzi) [inline Buy/Graph.Client.swift]: @sav007 ^^

File: Buy/Graph.RetryHandler.swift (hunk 1 above)
Comment (davidmuzi) [inline Buy/Graph.RetryHandler.swift]: should have a few unit tests

File: Buy/Graph.Client.swift (hunk 2 above)
Comment (dbart01) [inline Buy/Graph.Client.swift]: Is that **in addition** to the `User-Agent` header?

Comment (davidmuzi) [inline Buy/Graph.Client.swift]: yea, leave it in since its indexed in Splunk so easy to query

File: Buy/Graph.RetryHandler.swift (hunk 1 above)
Comment (dbart01) [inline Buy/Graph.RetryHandler.swift]: Comprehensive unit test **are** coming, just not in this PR.

Comment (davidmuzi) [inline Buy/Graph.RetryHandler.swift]: OK.  Prefer to have smaller PRs with Units tests though

File: Buy/Graph.Task.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public extension Graph {
+    
+    public class Task {
```
Comment (davidmuzi) [inline Buy/Graph.Task.swift]: Whats the intended purpose of this class?

Comment (dbart01) [inline Buy/Graph.Task.swift]: We need reference-type wrappers for all `URLSessionDataTask`s thanks to the retry handler. If a request is retried, the original `URLSessionDataTask` will be invalid and the user will have no way to cancel retried requests. With `Graph.Task`, all handles are managed internally and any `resume` or `cancel` calls are forwarded to the relevant handle.

Comment (davidmuzi) [inline Buy/Graph.Task.swift]: 👍

File: Buy/Graph.Client.swift (hunk 2 above)
Comment (dbart01) [inline Buy/Graph.Client.swift]: There's discussion surrounding whether this is necessary / details of implementation so I'll hold off including it in this PR.

**Pull request merged**